Upstreamed as
<https://gitlab.com/orcus/orcus/commit/05d3e608ce09a9cf34aa085de6180978a03f0f13>
"Allow passed-in CXXFLAGS to override -fvisibility=hidden":

--- configure
+++ configure
@@ -4255,7 +4255,7 @@
 
   fi
 
-CXXFLAGS="$CXXFLAGS -fvisibility=hidden -Wall -pthread"
+CXXFLAGS="-fvisibility=hidden $CXXFLAGS -Wall -pthread"
 
 # ===========================
 # Find required base packages
